### PR TITLE
Removed deprecated verify argument from jwt decode; updated PyJWT version

### LINKED ISSUE #10: Client incompatible with latest version of PyJWT (2.0.0)
I would say the solution is to either freeze the version of PyJWT or add the required argument to jwt.decode

Exception:
```python
---------------------------------------------------------------------------
DecodeError                               Traceback (most recent call last)
<ipython-input-2-b5a5ca0518b4> in <module>
     11 
     12 client = vtgclient.Client(HOST, PORT)
---> 13 client.authenticate(USERNAME, PASSWORD)
     14 
     15 client.setup_encryption(None)

/opt/conda/lib/python3.8/site-packages/vantage6/client/__init__.py in authenticate(self, username, password)
    455         # belongs. This is usefull for some client side checks
    456         type_ = "user"
--> 457         id_ = jwt.decode(self.token, verify=False)['identity']
    458         user = self.request(f"user/{id_}")
    459         name = user.get("firstname")

/opt/conda/lib/python3.8/site-packages/jwt/api_jwt.py in decode(self, jwt, key, algorithms, options, **kwargs)
    111         **kwargs,
    112     ) -> Dict[str, Any]:
--> 113         decoded = self.decode_complete(jwt, key, algorithms, options, **kwargs)
    114         return decoded["payload"]
    115 

/opt/conda/lib/python3.8/site-packages/jwt/api_jwt.py in decode_complete(self, jwt, key, algorithms, options, **kwargs)
     77 
     78         if options["verify_signature"] and not algorithms:
---> 79             raise DecodeError(
     80                 'It is required that you pass in a value for the "algorithms" argument when calling decode().'
     81             )

DecodeError: It is required that you pass in a value for the "algorithms" argument when calling decode().

```

### PR DESCRIPTION
**Pull request by bartvanb**
Fix #10

### CODE REVIEW

File: vantage6-client/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cryptography==3.3.2
 requests==2.25.1
-PyJWT==1.7.1
+PyJWT
```
Comment (frankcorneliusmartin) [inline vantage6-client/requirements.txt:3]: I would prefer to freeze the version

File: vantage6-client/setup.py [hunk 2]
```diff
@@ -36,7 +36,7 @@
     install_requires=[
         'cryptography==3.3.2',
         'requests==2.25.1',
-        'PyJWT==1.7.1',
+        'PyJWT',
```
Comment (frankcorneliusmartin) [inline vantage6-client/setup.py:39]: Same for this one

File: vantage6-node/setup.py [hunk 3]
```diff
@@ -51,7 +51,7 @@
             'SQLAlchemy==1.3.15',
             'schema==0.7.1',
             'appdirs==1.4.3',
-            'PyJWT==1.7.1',
+            'PyJWT',
```
Comment (frankcorneliusmartin) [inline vantage6-node/setup.py:54]: Aaaan this one

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]

**Comment by frankcorneliusmartin** [comment]
It seems something is not completely the same. This is from the server logs:
```python
Traceback (most recent call last):
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask\\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask\\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask_restful\\__init__.py", line 467, in wrapper
    resp = resource(*args, **kwargs)
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask\\views.py", line 84, in view
    return current_app.ensure_sync(self.dispatch_request)(*args, **kwargs)
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask_restful\\__init__.py", line 582, in dispatch_request
    resp = meth(*args, **kwargs)
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flasgger\\utils.py", line 273, in wrapper
    return function(*args, **kwargs)
  File "C:\\Users\\FMa1805.36838\\Repositories\\vantage6\\vantage6-server\\vantage6\\server\\resource\\token.py", line 101, in post
    token = create_access_token(user)
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask_jwt_extended\\utils.py", line 173, in create_access_token
    headers=headers)
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask_jwt_extended\\jwt_manager.py", line 522, in _create_access_token
    headers=headers
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask_jwt_extended\\tokens.py", line 77, in encode_access_token
    json_encoder=json_encoder, headers=headers)
  File "C:\\tools\\miniconda3\\envs\\work\\lib\\site-packages\\flask_jwt_extended\\tokens.py", line 30, in _encode_jwt
    json_encoder=json_encoder, headers=headers).decode(\'utf-8\')
AttributeError: \'str\' object has no attribute \'decode\'\
```

**Comment by frankcorneliusmartin** [comment]
Also on the side note: 

```
vantage6-common

requirements.txt has additional requirements when compared to the setup.py file.
wincertstore==0.2
contextlib2==0.5.5
certifi==2020.4.5.1
```

File: vantage6-server/tests_server/test_resources.py [hunk 4]
```diff
@@ -108,9 +109,11 @@ def login(self, type_='root'):
             headers = {
                 'Authorization': 'Bearer {}'.format(tokens['access_token'])
             }
+            return headers
         else:
             print('something wrong, during login:')
             print(tokens)
+            return None
 
         return headers
```
Comment (bartvanb) [inline vantage6-server/tests_server/test_resources.py:118]: I think this final return can now be deleted

File: vantage6-server/vantage6/server/__init__.py [hunk 5]
```diff
@@ -219,25 +219,25 @@ def remove_db_session(response):
...
-        def error_remove_db_session(error):
+        def error_remove_db_session(error: HTTPException):
             """In case an HTTP-exception occurs during the request.
 
             It is important to close the db session to avoid having dangling
             sessions.
             """
-            log.warn('Error occured during request')
+            log.warn('HTTP Exception occured during request')
             log.debug(traceback.format_exc())
```
Comment (bartvanb) [inline vantage6-server/vantage6/server/__init__.py:229]: We should consider turning this into `log.exception(error)` (and same 10 lines below)

**Review by bartvanb — COMMENTED** [review]
Looks good. Few additional suggestions.

File: Makefile [hunk 6]
```diff
@@ -110,3 +110,6 @@ clean:
 	cd vantage6 && make clean
 	cd vantage6-node && make clean
 	cd vantage6-server && make clean
+
+test:
+	coverage run --source=vantage6 --omit="utest.py","*.html","*.htm","*.txt","*.yml","*.yaml" utest.py
```
[the remark this answers is not in the document]
Comment (frankcorneliusmartin) [reply Makefile:115]: It already runs the unittests?

File: vantage6-client/requirements.txt [hunk 7]
```diff
@@ -1,6 +1,7 @@
 cryptography==3.3.2
 requests==2.25.1
-PyJWT==1.7.1
-vantage6-common >= 1.0.0b13
+PyJWT==2.4.0
 pyfiglet==0.8.post1
-SPARQLWrapper==1.8.5
\ No newline at end of file
+SPARQLWrapper==1.8.5
+pandas
```
[the remark this answers is not in the document]
Comment (frankcorneliusmartin) [reply vantage6-client/requirements.txt:6]: I do agree, however it is related to the pandas version in the algorithm container (or better the client version within the container). 

-> Seperate issue: https://github.com/vantage6/vantage6/issues/266

File: vantage6-server/tests_server/test_resources.py (hunk 4 above)
Comment (frankcorneliusmartin) [reply vantage6-server/tests_server/test_resources.py:118]: Yes, I dont know why its still there :')

File: vantage6-server/vantage6/server/__init__.py (hunk 5 above)
Comment (frankcorneliusmartin) [reply vantage6-server/vantage6/server/__init__.py:229]: Yes!

**Pull request merged**